libsolidity/codegen/ir/IRGenerator.h

```diff
@@ -67,12 +67,21 @@ class IRGenerator
 	/// Generates code that assigns the initial value of the respective type.
 	std::string generateInitialAssignment(VariableDeclaration const& _varDecl);
 
+	/// generates implicit constructor for all base contracts including itself
+	/// if there is user defined constructor, it will called from implicit constructor
+	void generateImplicitConstructors(ContractDefinition const& _contract);
+	std::pair<std::string, std::map<ContractDefinition const*, std::string>> evaluateConstructorArguments(
+		ContractDefinition const& _contract
+	);
+	std::string initStateVariables(ContractDefinition const& _contract);
```

Comment:
Please document.

The main source of confusion for these functions is the question whether it only concerns the contract that is passed or the whole hierarchy.